stan/math/opencl/matrix_cl.hpp

```diff
@@ -4,15 +4,18 @@
 #include <stan/math/opencl/opencl_context.hpp>
 #include <stan/math/opencl/kernel_cl.hpp>
 #include <stan/math/opencl/constants.hpp>
+#include <stan/math/opencl/cache_copy.hpp>
 #include <stan/math/prim/mat/fun/Eigen.hpp>
+#include <stan/math/prim/arr.hpp>
 #include <stan/math/prim/scal/err/check_size_match.hpp>
 #include <stan/math/prim/scal/err/domain_error.hpp>
 #include <stan/math/opencl/kernels/copy.hpp>
 #include <stan/math/opencl/kernels/sub_block.hpp>
+#include <stan/math/opencl/kernels/unpack.hpp>
 #include <stan/math/opencl/kernels/triangular_transpose.hpp>
 #include <stan/math/opencl/kernels/zeros.hpp>
+#include <stan/math/rev/scal/fun/value_of_rec.hpp>
```

Comment:
What happens to the failing test if you remove this include? (and I guess the code that depends on it)

Reply:
That didnt help. But you gave me an idea. `#include <stan/math/prim/arr.hpp>` is the culprit here. Removing that and the code that depends on it, makes the tests pass.

Reply:
We need to be careful - the opencl stuff has so far been outside of our scal/arr/mat prim/rev/fwd/mix taxonomy, but it seems like we still need to be mindful of it, at least until the flattening happens for scal/arr/mat (this won't help us with prim/rev/fwd/mix yet). We might need to think about separating some of this stuff out into those folders...

Reply:
For this PR I resolved this with a specialization for std::vector<double>. I am not really sure how we would separate this stuff out..

Anyways, this is now ready for another review when you have the time.